### DISCUSSION TITLE
Simplify navigation and initialize dependency

### DISCUSSION THREAD

**Discussion by bugix** [pointfreeco/swift-composable-architecture#2080] [Q&A]
I prototyped some code [here](https://gist.github.com/bugix/ffd435c9c61525f0768990429daec543) which should ask the user for a token and navigate to a second feature which uses a dependency that relies on this token.

My questions are:

1. Can the navigation be simplified by eg. removing the Destination Reducer? Is there an episode discussing this?
2. How can the user provided token be injected to the ApiClient dependency? I read in another discussion, that the not yet landed ReducerReader will do that, but I fail to place it in my code.

**Comment by mariob** [comment]
Hi @bugix 

Regards 1 - You could use SwitchStore and an Enum State instead of _traditional_ navigation. This will just switch the content of the view body. You can add any transition you want when the views are switched. I use it myself for the "Login -> Main" flow in an app.

Check out the announcement of SwitchStore here:  [https://www.pointfree.co/blog/posts/59-announcing-switchstore-for-the-composable-architecture](https://www.pointfree.co/blog/posts/59-announcing-switchstore-for-the-composable-architecture)

Example code here: [https://github.com/pointfreeco/swift-composable-architecture/tree/main/Examples/TicTacToe/tic-tac-toe/Sources](https://github.com/pointfreeco/swift-composable-architecture/tree/main/Examples/TicTacToe/tic-tac-toe/Sources).

Regarding 2 - I'd like to see the ReducerReader be added to the framework (or whatever it will be called) because then I'll probably add some _session data_ as dependency when the user logs in. Currently I use a dependency with a CurrentValueSubject which I convert to an AsyncStream for monitoring what happens to the session. I guess that won't go away even with injecting a dependency with session data.
